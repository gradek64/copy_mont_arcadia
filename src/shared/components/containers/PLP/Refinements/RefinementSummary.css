/** @define RefinementSummary */
.RefinementSummary {
  display: block;
  padding-bottom: 15px;

  &-header{
    align-items: center;
    display: flex;
    justify-content: space-between;
    border-bottom: $refinements-summary-header-border-bottom;
    height: $plp-filters-height;
    padding: $plp-filters-spacing-vertical 0;
    padding-top: $plp-filters-padding-top;
    box-sizing: content-box;
  }
  &-headerTitle {
    display: inline-block;
    font-size: $refinements-summary-header-size;
    color: $refinements-summary-header-color;
    margin-bottom: $plp-filters-spacing-vertical;
    margin-top: $plp-filters-spacing-vertical;
    font-weight: $refinements-summary-header-font-weight;
  }
  &-clearRefinementsButton {
    position: relative;
    font-size: $refinements-summary-item-size;
    text-decoration: $clearall-text-decoration;

    &:disabled {
      color: $colorFaded;
      cursor: default;
    }
  }
  &-categoryItem {
    margin-top: $refinements-summary-category-item-margin-top;
    text-transform: capitalize;
  }
  &-item {
    margin-top: $refinements-summary-item-margin-top;
    text-transform: capitalize;
  }
  &-itemTitle {
    margin-bottom: $refinements-summary-item-title-margin-bottom;
    font-size: $refinements-summary-item-title-size;
    font-weight: $refinements-summary-item-title-font-weight;
    display: block;
    color: $refinements-summary-item-title-color;
  }
  &-valuesContainer {
    display: flex;
    flex-flow: row wrap;
  }
   /* Deprecated temporary (replaced by removeTextValue, to fix I11 shape not supported DES-3000) */
  &-removeValue {
    width: $refinements-summary-item-remove-icon-size;
    height: $refinements-summary-item-remove-icon-size;
    min-height: $refinements-summary-item-remove-icon-size;
    margin-left: 5px;
    cursor: pointer;
    background-color: $refinements-summary-item-color;
    mask: url("public/$brand/images/close-icon.svg") no-repeat center;
    mask-size: $refinements-summary-item-remove-icon-size;
  }
  &-removeTextValue {
     margin-left: 5px;
     cursor: pointer;
     color: $refinements-summary-item-color;
   }
  &-value {
    font-size: $refinements-summary-item-size;
    background-color: $refinements-summary-item-background-color;
    padding: $refinements-summary-item-padding;
    color: $refinements-summary-item-color;
    border: $refinements-summary-item-border;
    margin-right: 5px;
    margin-bottom: 5px;
    display: inline-block;
  }
  &-refinementValue {
    display: flex;
    align-items: center;
  }
}

/* postcss-bem-linter: ignore */
.RefinementSummary-removeTextValue:hover {
  color: $refinements-summary-item-remove-icon-hover;
}
/* Deprecated temporary (replaced by removeTextValue, to fix I11 shape not supported DES-3000) */
/* postcss-bem-linter: ignore */
.no-touch .RefinementSummary-removeValue:enabled:hover {
  background-color: $refinements-summary-item-remove-icon-hover;
}

@media $min-tablet {
  .RefinementSummary {
    &-header{
      height: $plp-filters-height-tablet;
    }
    &-valuesContainer {
      flex-flow: column wrap;
    }
    &-value {
      margin-right: 0;
      position: relative;
      padding-bottom: $refinements-summary-value-padding-bottom-desktop;
    }
    &-refinementValue {
      padding-right: $refinements-summary-value-padding-right-desktop;
      display: inline-block;
    }
    &-valueInner {
      height: $refinements-summary-item-height-desktop;
    }
    &-removeValue {
      position: absolute;
      top: $refinements-summary-item-top-desktop;
      bottom: 0;
      right: $refinements-summary-item-right-desktop;
      margin: auto 0;
    }
  }
}

@media $min-laptop {
  .RefinementSummary {
    &-header{
      height: $plp-filters-height;
    }
  }
}
