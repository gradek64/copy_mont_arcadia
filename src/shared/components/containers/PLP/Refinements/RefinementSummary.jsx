/* eslint-disable jsx-a11y/interactive-supports-focus */
import React, { Component } from 'react'
import { browserHistory } from 'react-router'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

// Utils/Helpers
import {
  updateSeoUrlIfSearchFilter,
  isSeoUrlSearchFilter,
  getResetSearchUrl,
  removePriceFromSeoUrl,
  addParamClearAll,
} from '../../../../lib/products-utils'

import { sendAnalyticsFilterUsedEvent } from '../../../../analytics/analytics-actions'

// selectors
import {
  getActiveRefinements,
  getRemoveAllFilters,
} from '../../../../selectors/refinementsSelectors'
import { getCanonicalUrl } from '../../../../selectors/navigationSelectors'
import { getRouteSearch } from '../../../../selectors/routingSelectors'
import { getCurrentSortOption } from '../../../../selectors/productSelectors'

// components
import Price from '../../../common/Price/Price'

@connect(
  (state) => ({
    activeRefinements: getActiveRefinements(state),
    categoryTitle: state.products.categoryTitle,
    currentSearchPath: getRouteSearch(state),
    currentSortOptions: getCurrentSortOption(state),
    removeAllFilters: getRemoveAllFilters(state),
    canonicalUrl: getCanonicalUrl(state),
  }),
  {
    sendAnalyticsFilterUsedEvent,
  }
)
class RefinementSummary extends Component {
  static propTypes = {
    activeRefinements: PropTypes.array,
    categoryTitle: PropTypes.string.isRequired,
    currentSearchPath: PropTypes.string,
    currentSortOptions: PropTypes.object,
    removeAllFilters: PropTypes.string,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  static defaultProps = {
    categoryTitle: '',
    activeRefinements: [],
  }

  clearRefinements = () => {
    const {
      sendAnalyticsFilterUsedEvent,
      currentSearchPath,
      removeAllFilters,
      canonicalUrl,
    } = this.props
    if (isSeoUrlSearchFilter(removeAllFilters)) {
      // if is search go-to : /search?q=red
      browserHistory.push(getResetSearchUrl(currentSearchPath))
    } else {
      // if is category go-to : seUrl (removeAllFilters)
      const urlToUse = () => {
        if (removeAllFilters && removeAllFilters.includes(canonicalUrl)) {
          return removeAllFilters
        }
        return canonicalUrl
      }
      browserHistory.push(addParamClearAll(urlToUse()))
    }

    sendAnalyticsFilterUsedEvent({
      filterCategory: 'clear all',
      filterOption: 'clear all',
      filterAction: 'remove',
    })
  }

  removeRefinement = (key, seoUrl) => {
    const { currentSortOptions } = this.props
    if (key === 'NOWPRICE') {
      browserHistory.push(
        updateSeoUrlIfSearchFilter(
          removePriceFromSeoUrl(currentSortOptions.navigationState)
        )
      )
    } else {
      browserHistory.push(updateSeoUrlIfSearchFilter(seoUrl))
    }
  }

  renderRefinement = (refinement) => {
    const { l } = this.context
    if (refinement.title !== l`Category`) {
      const values =
        refinement.values &&
        refinement.values.map((value) => {
          return (
            <span key={value.key} className="RefinementSummary-valueWrapper">
              <span className="RefinementSummary-value RefinementSummary-refinementValue">
                <span className="RefinementSummary-valueInner">
                  {refinement.key === 'NOWPRICE' ? (
                    <div>
                      <Price price={value.lowerBound} /> -{' '}
                      <Price price={value.upperBound} />
                    </div>
                  ) : (
                    value.label
                  )}
                </span>
                <a
                  className="RefinementSummary-removeTextValue"
                  role="button"
                  onClick={() =>
                    this.removeRefinement(refinement.key, value.seoUrl)
                  }
                >
                  X
                </a>
              </span>
            </span>
          )
        })
      return (
        <div key={refinement.key} className="RefinementSummary-item">
          <span className="RefinementSummary-itemTitle">
            {refinement.title}
          </span>
          <div className="RefinementSummary-valuesContainer">{values}</div>
        </div>
      )
    }
  }

  // disable clear all button
  disableClearAllButton = () => {
    const { removeAllFilters, activeRefinements } = this.props

    return isSeoUrlSearchFilter(removeAllFilters)
      ? activeRefinements.length === 0
      : activeRefinements.length < 2
  }

  render() {
    const { categoryTitle, activeRefinements } = this.props
    const { l } = this.context
    return (
      <div className="RefinementSummary">
        <header className="RefinementSummary-header">
          <span className="RefinementSummary-headerTitle">{l`Filter`}</span>
          <button
            disabled={this.disableClearAllButton()}
            className={'RefinementSummary-clearRefinementsButton'}
            onClick={this.clearRefinements}
          >{l`Clear all`}</button>
        </header>
        <div className="RefinementSummary-categoryItem">
          <span className="RefinementSummary-itemTitle">{l`Category`}</span>
          {categoryTitle && (
            <span className="RefinementSummary-value">{l(categoryTitle)}</span>
          )}
        </div>
        {activeRefinements.map((refinement) =>
          this.renderRefinement(refinement)
        )}
      </div>
    )
  }
}

export default RefinementSummary
