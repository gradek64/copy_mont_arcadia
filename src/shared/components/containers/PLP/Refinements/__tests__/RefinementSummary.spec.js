import testComponentHelper from 'test/unit/helpers/test-component'
import RefinementSummary from '../RefinementSummary'
import { browserHistory } from 'react-router'

jest.mock('react-router', () => ({
  browserHistory: {
    push: jest.fn(),
  },
}))

describe('<RefinementSummary/>', () => {
  const renderComponent = testComponentHelper(
    RefinementSummary.WrappedComponent
  )
  const defaultProps = {
    products: {
      sortOptions: {},
      products: [],
    },
    removeAllFilters:
      '/en/tsuk/category/shoes-430/N-8ewZdgl?Nrpp=24&siteId=%2F12556&categoryId=null',
    categoryTitle: 'Dress',
    canonicalUrl: '/en/tsuk/category/shoes-430',
  }
  const activeRefinements = [
    {
      key: 'TOPSHOP_UK_CE3_PRODUCT_TYPE',
      title: 'Product Type',
      values: [
        {
          key: 'ECMC_PROD_CE3_PRODUCT_TYPE_1_DRESSES',
          label: 'dresses',
          seoUrl: 'XXXXX',
        },
      ],
    },
    {
      key: 'NOWPRICE',
      title: 'Price',
      values: [
        {
          key: 'NOWPRICE5.032.0',
          label: '',
          upperBound: '32.0',
          lowerBound: '5.0',
        },
      ],
    },
    {
      key: 'TOPSHOP_UK_CATEGORY',
      title: 'Category',
      values: [
        {
          key: 3552,
          label: 'Dress',
        },
      ],
    },
  ]

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(defaultProps).getTree()).toMatchSnapshot()
    })
    it('with activeRefinements', () => {
      expect(
        renderComponent({ ...defaultProps, activeRefinements }).getTree()
      ).toMatchSnapshot()
    })

    it('Should not display Refinements value block under the category if the value is empty', () => {
      const { wrapper } = renderComponent({
        ...defaultProps,
        categoryTitle: '',
      })
      const refinementsSummaryValue = wrapper.find('.RefinementSummary-value')
      expect(refinementsSummaryValue).toHaveLength(0)
    })
  })

  describe('@events', () => {
    it('removes a refinement option value', () => {
      const { wrapper } = renderComponent({
        ...defaultProps,
        categoryTitle: 'Dress',
        activeRefinements,
      })
      wrapper
        .find('.RefinementSummary-removeTextValue')
        .at(0)
        .prop('onClick')()
      expect(browserHistory.push).not.toHaveBeenCalledWith(['XXXXX'])
    })

    it('when clear refinements button is clicked', () => {
      const { wrapper } = renderComponent({
        ...defaultProps,
        activeRefinements,
        sendAnalyticsFilterUsedEvent: jest.fn(),
      })
      const clearRefinementsButton = wrapper.find(
        '.RefinementSummary-clearRefinementsButton'
      )
      clearRefinementsButton.simulate('click')
      expect(browserHistory.push).toHaveBeenCalled()
      expect(browserHistory.push).toHaveBeenCalledWith(
        '/en/tsuk/category/shoes-430/N-8ewZdgl?Nrpp=24&siteId=%2F12556&categoryId=null&clearAll=true'
      )
    })

    it('when clear refinements button is clicked WHIT SEARCH RESULT', () => {
      const { instance, wrapper } = renderComponent({
        ...defaultProps,
        removeAllFilters: '?seo=false&siteId=%2F12556',
        currentSearchPath: '?Nrpp=24&Ntt=red&seo=false&siteId=%2F12556',
        activeRefinements,
        sendAnalyticsFilterUsedEvent: jest.fn(),
      })
      const clearRefinementsButton = wrapper.find(
        '.RefinementSummary-clearRefinementsButton'
      )
      expect(instance.disableClearAllButton()).toBe(false)
      clearRefinementsButton.simulate('click')
      expect(browserHistory.push).toHaveBeenCalledWith('/search/?q=red')
    })

    it('clear button should disabled', () => {
      const { instance } = renderComponent({
        ...defaultProps,
        removeAllFilters: '/N-275aZdgl?Nrpp=24&siteId=%2F12556&categoryId=null',
        activeRefinements: [],
        sendAnalyticsFilterUsedEvent: jest.fn(),
      })
      expect(instance.disableClearAllButton()).toBe(true)
    })
  })
})
