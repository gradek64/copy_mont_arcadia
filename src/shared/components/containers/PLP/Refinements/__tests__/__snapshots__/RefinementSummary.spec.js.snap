// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RefinementSummary/> @renders in default state 1`] = `
<div
  className="RefinementSummary"
>
  <header
    className="RefinementSummary-header"
  >
    <span
      className="RefinementSummary-headerTitle"
    >
      Filter
    </span>
    <button
      className="RefinementSummary-clearRefinementsButton"
      disabled={true}
      onClick={[Function]}
    >
      Clear all
    </button>
  </header>
  <div
    className="RefinementSummary-categoryItem"
  >
    <span
      className="RefinementSummary-itemTitle"
    >
      Category
    </span>
    <span
      className="RefinementSummary-value"
    >
      Dress
    </span>
  </div>
</div>
`;

exports[`<RefinementSummary/> @renders with activeRefinements 1`] = `
<div
  className="RefinementSummary"
>
  <header
    className="RefinementSummary-header"
  >
    <span
      className="RefinementSummary-headerTitle"
    >
      Filter
    </span>
    <button
      className="RefinementSummary-clearRefinementsButton"
      disabled={false}
      onClick={[Function]}
    >
      Clear all
    </button>
  </header>
  <div
    className="RefinementSummary-categoryItem"
  >
    <span
      className="RefinementSummary-itemTitle"
    >
      Category
    </span>
    <span
      className="RefinementSummary-value"
    >
      Dress
    </span>
  </div>
  <div
    className="RefinementSummary-item"
    key="TOPSHOP_UK_CE3_PRODUCT_TYPE"
  >
    <span
      className="RefinementSummary-itemTitle"
    >
      Product Type
    </span>
    <div
      className="RefinementSummary-valuesContainer"
    >
      <span
        className="RefinementSummary-valueWrapper"
        key="ECMC_PROD_CE3_PRODUCT_TYPE_1_DRESSES"
      >
        <span
          className="RefinementSummary-value RefinementSummary-refinementValue"
        >
          <span
            className="RefinementSummary-valueInner"
          >
            dresses
          </span>
          <a
            className="RefinementSummary-removeTextValue"
            onClick={[Function]}
            role="button"
          >
            X
          </a>
        </span>
      </span>
    </div>
  </div>
  <div
    className="RefinementSummary-item"
    key="NOWPRICE"
  >
    <span
      className="RefinementSummary-itemTitle"
    >
      Price
    </span>
    <div
      className="RefinementSummary-valuesContainer"
    >
      <span
        className="RefinementSummary-valueWrapper"
        key="NOWPRICE5.032.0"
      >
        <span
          className="RefinementSummary-value RefinementSummary-refinementValue"
        >
          <span
            className="RefinementSummary-valueInner"
          >
            <div>
              <Price
                className=""
                price="5.0"
              />
               -
               
              <Price
                className=""
                price="32.0"
              />
            </div>
          </span>
          <a
            className="RefinementSummary-removeTextValue"
            onClick={[Function]}
            role="button"
          >
            X
          </a>
        </span>
      </span>
    </div>
  </div>
</div>
`;
